### DISCUSSION TITLE
"Error": "API error: API error Error(InternalError): Failed to convert transaction data from storage

### DISCUSSION THREAD

**Discussion by Aladeenb** [aptos-labs/aptos-developer-discussions#109] [Questions] [answered]
`aptos move publish` fails with the following error:

```zsh
  "Error": "API error: API error Error(InternalError): 
  Failed to convert transaction data from storage: 
  Module ModuleId { 
   address: <module-address>, 
   name: Identifier(\"<resource-name>\") 
  } can't be found"
```

## How to reproduce

1. clone `https://github.com/code-and-ledger/super-memory.git` .
2. Initialise local configuration.
3. attempt to publish the `SCENARIO 2`.

## Notes

- The difference between scenario 1 and scenario
2 is number of functions involved (see comments).

- A similar issue is this: https://github.com/aptos-labs/aptos-core/issues/11038

**Comment by gregnazario** [comment, marked as answer]
Yes, looks like that bug applies.

The first situation, you're creating a coin, which is not emitting a module event.

The second situation, you're creating the coin, and adding liquidity.  This likely is creating the issue when an event is created on transfer of coins.

There are a couple bugs like this one around `init_module`, and usually when I write my own contracts, I'll avoid using `init_module`.  I'll only use it if it's simple enough.

Otherwise, I'll usually add an `initialize()` function, and call that after publishing.

**Reply by Aladeenb** [reply]
Thanks for the quick response. 

I thought of this solution as well. 

However, this will require two transactions instead of only one. 

Is it possible to publish a module via Move scripts? If yes, I think I will be able to deploy using code.move and then call the initialise function

**Reply by gregnazario** [reply]
Yeah, it should be possible, but I think you might still run into the same error.

I think it has to deal with a VM issue on emitting an event in the same transaction as the code is published cc @lightmark
